src/Workspaces/Core/Portable/Workspace/Solution/Document.cs

```diff
@@ -456,17 +462,18 @@ public async Task<IEnumerable<TextChange>> GetTextChangesAsync(Document oldDocum
                 // get changes by diffing the trees
                 if (this.SupportsSyntaxTree)
                 {
-                    var tree = (await this.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false))!;
-                    var oldTree = await oldDocument.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+                    var tree = this.GetSyntaxTreeSynchronously(cancellationToken);
```

Comment:
@CyrusNajmabadi Not sure I understand this bit unless you had some other data for it -- GetSyntaxTreeSynchronously will block threads if we have to read the tree from disk. That's considered acceptable in the UI thread case (since the thread is blocked either way), but this _could_ create problems if we we're calling it on a bunch of threads all at once since we might block up the thread pool.